images/virtualization-artifact/pkg/apiserver/server/config.go

```diff
@@ -0,0 +1,83 @@
+package server
+
+import (
+	"errors"
+	"fmt"
+
+	genericapiserver "k8s.io/apiserver/pkg/server"
+	"k8s.io/client-go/rest"
+
+	virtv2 "github.com/deckhouse/virtualization-controller/api/core/v1alpha2"
+	"github.com/deckhouse/virtualization-controller/pkg/apiserver/api"
+	rest2 "github.com/deckhouse/virtualization-controller/pkg/apiserver/registry/vm/rest"
+	"github.com/deckhouse/virtualization-controller/pkg/tls/certManager/filesystem"
+)
+
+var ErrConfigInvalid = errors.New("configuration is invalid")
+
+type Config struct {
+	Apiserver           *genericapiserver.Config
+	Rest                *rest.Config
+	Kubevirt            rest2.KubevirtApiServerConfig
+	ProxyClientCertFile string
+	ProxyClientKeyFile  string
+}
+
+func (c Config) Validate() []error {
+	errs := []error{}
+	if c.Kubevirt.Endpoint == "" {
+		errs = append(errs, fmt.Errorf(".Kubevirt.Endpoint is required. %w", ErrConfigInvalid))
+	}
+	if c.Kubevirt.CaBundlePath == "" {
+		errs = append(errs, fmt.Errorf(".Kubevirt.CaBundlePath is required. %w", ErrConfigInvalid))
+	}
+	if c.Kubevirt.ServiceAccount.Name == "" {
+		errs = append(errs, fmt.Errorf(".Kubevirt.ServiceAccount.Name is required. %w", ErrConfigInvalid))
+	}
+	if c.Kubevirt.ServiceAccount.Namespace == "" {
+		errs = append(errs, fmt.Errorf(".Kubevirt.ServiceAccount.Namespace is required. %w", ErrConfigInvalid))
+	}
+	if c.ProxyClientCertFile == "" {
+		errs = append(errs, fmt.Errorf(".ProxyClientCertFile is required. %w", ErrConfigInvalid))
+	}
+	if c.ProxyClientKeyFile == "" {
+		errs = append(errs, fmt.Errorf(".ProxyClientKeyFile is required. %w", ErrConfigInvalid))
+	}
+	if c.Apiserver == nil {
+		errs = append(errs, fmt.Errorf(".Apiserver is required. %w", ErrConfigInvalid))
+	}
+	if c.Rest == nil {
+		errs = append(errs, fmt.Errorf(".Rest is required. %w", ErrConfigInvalid))
+	}
+	return errs
+}
+
+func (c Config) Complete() (*server, error) {
```

Comment:
Public method returns private struct

Reply:
Yes. It is assumed that you can create a server only server.Config.

Reply:
I agree with @Isteb4k. The are no reasons to return private from public constructor. For example, there is no such issue with FileCertificateManager in images/virtualization-artifact/pkg/tls/certManager/filesystem/file-cert-manager.go.